### ISSUE TITLE
[Bug]: Cannot search in users management page with global search enabled

### ISSUE THREAD

**Issue by blizzz** [nextcloud/server#41484]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

With global search enabled and on the users page, the list of users cannot be filtered anymore. When clicking "people" on the global search, and looking for a users, they are not found either. Also the duplicated input box is confusing.

[global search vs user mgmt.webm](https://github.com/nextcloud/server/assets/2184312/7eca8aa1-d53a-4b59-a712-d6e024f6f247)

### Steps to reproduce

1. Have global search enabled
2. go to users page
3. try to search for a user

### Expected behavior

A search for the entered string is performed and only matching users are shown. As it happens with old search.

### Installation method

None

### Nextcloud Server version

master

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by blizzz** [comment]
cc @Fenn-CS @Altahrim

**Comment by Altahrim** [comment]
It seems the user search provider is a bit incomplete: https://github.com/nextcloud/server/blob/master/apps/settings/lib/Search/UserSearch.php#L72-L75

**Comment by blizzz** [comment]
> It seems the user search provider is a bit incomplete: https://github.com/nextcloud/server/blob/master/apps/settings/lib/Search/UserSearch.php#L72-L75

Related comment: https://github.com/nextcloud/server/issues/41381#issuecomment-1812127225

**Comment by blizzz** [comment]
> It seems the user search provider is a bit incomplete: https://github.com/nextcloud/server/blob/master/apps/settings/lib/Search/UserSearch.php#L72-L75

Related comment: https://github.com/nextcloud/server/pull/40413#discussion_r1354628540

**Comment by AndyScherzinger** [comment]
Yeah, the user search works slightly different and doesn't forward results into the unified search or new advanced search.

Can you still trigger the original behavior for user searches @Altahrim @Fenn-CS

**Comment by AndyScherzinger** [comment]
* Only show "USERS" as a app-filter when being on the user-management page
* emit the unified search event

**Comment by ChristophWurst** [comment]
App management search doesn't work either. Same issue?

**Comment by Altahrim** [comment]
> App management search doesn't work either. Same issue?

Should be fixed with https://github.com/nextcloud/server/pull/41669. Not the best way but it will search in background.

**Comment by ChristophWurst** [comment]
> Should be fixed with #41669.

I still can't search on latest master

**Comment by blizzz** [comment]
Searching works now on RC2 albeit the UX is suboptimal. The modal is shown and active, and also gets search results from other sources, but the actual user search results appear in the background.

**Comment by AndyScherzinger** [comment]
So yeah, fixed for the moment while UX is not optimal, yet it works the way it did before except the modal is larger than the former popup.
